docs/_blog/bdk_rn_making_of.md

```diff
@@ -0,0 +1,238 @@
+
+```kotlin
+val descriptor: String = "wpkh(" + keys.xprv + "/84'/1'/0'/0/*)"
+
+val changeDescriptor: String = descriptor.replace("/84'/1'/0'/0/*","/84'/1'/0'/1/*")
+```
+
+To create a wallet with bdk we need to specify wallet descriptor, network, a database config, blockchain config. We intend to use bitcoin testnet and want to use default memory for data. For bitcoin node we will use a public electrum server. We will need to define these parameters to create a wallet.
+
+```kotlin
+val network = `Network.TESTNET`
+val databaseConfig = DatabaseConfig.Memory
+blockchainConfig =
+      BlockchainConfig.Electrum(
+          ElectrumConfig("ssl://electrum.blockstream.info:60002", null, 5u, null, 10u)
+      )
+```
+
+Once done we can use these parameters to create a bdk wallet using the native android bdk library:
+
+```kotlin
+var wallet: BdkWallet = BdkWallet(
+          descriptor,
+          changeDescriptor,
+          setNetwork(network),
+          databaseConfig,
+          config
+)
+```
+
+Once we have a wallet initialised, we can call methods on it to sync, generate a new address and to get balance 
+
+```kotlin
+wallet.sync(ProgressLog, maxAddress)
+
+wallet.getNewAddress()
+
+wallet.getBalance().toLong()
+```
+
+To return a value from the native android code to React Native’s Javascript side over the JS Native bridge we will use `com.facebook.react.bridge.Promise`. To return balance information to JS, the following code can be used
+
+```kotlin
+val balance: String = wallet.getBalance().toLong()
+result.resolve(balance)
+```
+
+At this point  we have an Android native module and it can be invoked from JS by calling createWallet. if the wallet is created correctly it will return the balance.
+
+This project can be imported into any RN project to reuse the defined methods without the need to carry out the setup described above.
+
+```javascript
+// any js file in React Native
+import BdkRn from 'bdk-rn';
+
+// create a wallet and retrieve balance
+const balance = await BdkRn.createWallet();
+console.log({ balance })
+
+```
+
+The actual bdk-rn module has organised the native code into granular methods for different stages of creating a wallet and for different interactions with the wallet, like generating seeds, creating wallet for different network, creating descriptors, creating or restoring wallet, fetching balance, fetching transactions  and many other methods. Please refer:[https://github.com/LtbLightning/bdk-rn#library-api](https://github.com/LtbLightning/bdk-rn#library-api) The set of APIs available will grow in the near future as more APIs will get added. This article can also be used as a guide to add new methods to the existing bdk-rn project.
+
+The objective of `bdk-rn` is to enable React Native developers to quickly start developing applications without the need to package BDK as described above. This article can also be used as a guide to add new methods to the existing bdk-rn project.
```

Comment:
This sentence is repeated from above (one of the two should be deleted): _This article can also be used as a guide to add new methods to the existing bdk-rn project._

Reply:
Done ✅